core/src/main/java/io/undertow/server/handlers/ProxyPeerAddressHandler.java

```diff
@@ -92,16 +92,18 @@ public void handleRequest(HttpServerExchange exchange) throws Exception {
                     value = value.substring(0, index);
                 }
             }
-            int port = 80;
+            int port = 0;
+            String hostHeader = NetworkUtils.formatPossibleIpv6Address(value);
             if(forwardedPort != null) {
                 try {
                     port = Integer.parseInt(forwardedPort);
+                    hostHeader += ":" + port;
```

Comment:
This is cherry-picked from commit: https://github.com/undertow-io/undertow/commit/a5ae059b824bd67e22e16a5e12c77293c0753a23 from 1.4.x branch.

But I think if the `forwardedPort != null`, then the port should be included in the `hostHeader`.  The `forwardedPort = exchange.getRequestHeaders().getFirst(Headers.X_FORWARDED_PORT)`